test/unit/TracerPerpetualSwaps.js

```diff
@@ -0,0 +1,267 @@
+const { expect } = require("chai")
+const { ethers, getNamedAccounts, deployments } = require("hardhat")
+const { deploy } = deployments
+const { smockit } = require("@eth-optimism/smock")
+const { BigNumber } = require("ethers")
+const zeroAddress = "0x0000000000000000000000000000000000000000"
+
+// create hardhat optimised feature
+const setup = deployments.createFixture(async () => {
+    // todo setup and mock appropriate functions
+})
+
+describe("Unit tests: Insurance.sol", function () {
+    beforeEach(async function () {
+        // todo call setup
+    })
+
+    describe("deposit", async () => {
```

Comment:
What about testing `deposit` when the amount is too high (i.e. `amount` WADified goes over `MAX_INT256`)?

Reply:
@BenjaminPatch I'm thinking this might be part of unit tests for LibBalances? Since it should test that those functions revert on overflow